openedx/core/djangoapps/credit/tests/test_verification_access.py

```diff
@@ -0,0 +1,270 @@
+"""
+Tests for in-course reverification user partition creation.
+
+This should really belong to the verify_student app,
+but we can't move it there because it's in the LMS and we're
+currently applying these rules on publish from Studio.
+
+In the future, this functionality should be a course transformation
+defined in the verify_student app, and these tests should be moved
+into verify_student.
+
+"""
+
+from mock import patch
+
+from django.conf import settings
+
+from openedx.core.djangoapps.credit.models import CreditCourse
+from openedx.core.djangoapps.credit.partition_schemes import VerificationPartitionScheme
+from openedx.core.djangoapps.credit.verification_access import create_verification_partitions
+from openedx.core.djangoapps.credit.signals import on_pre_publish
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import SignalHandler
+from xmodule.modulestore.exceptions import ItemNotFoundError
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, TEST_DATA_SPLIT_MODULESTORE
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory, check_mongo_calls_range
+from xmodule.partitions.partitions import Group, UserPartition
+
+
+class CreateVerificationPartitionTest(ModuleStoreTestCase):
+    """
+    Tests for applying verification access rules.
+    """
+
+    # Run the tests in split modulestore
+    # While verification access will work in old-Mongo, it's not something
+    # we're committed to supporting, since this feature is meant for use
+    # in new courses.
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @patch.dict(settings.FEATURES, {"ENABLE_COURSEWARE_INDEX": False})
```

Comment:
Is this being turned-off to prevent indexing when the course content is modified? If that is the case, can we disable this be default and only enable it in the specific tests that need it?